dd-sdk-android-gradle-plugin/src/test/kotlin/com/datadog/gradle/plugin/DdAndroidGradlePluginFunctionalTest.kt

```diff
@@ -0,0 +1,323 @@
+
+        // Given
+        stubGradleBuildFromResourceFile("build_with_check_deps_disabled.gradle.kts")
+
+        // When
+        val result = GradleRunner.create()
+            .withProjectDir(testProjectDir)
+            .withArguments("--configuration-cache", "assembleDebug")
+            .withPluginClasspath(getTestConfigurationClasspath())
+            .build()
+
+        // Then
+        assertThat(result.task(":assembleDebug")?.outcome).isEqualTo(TaskOutcome.SUCCESS)
+    }
+
+    @Test
+    fun `M success W assemble { Datadog SDK not in deps, checkProjectDepedencies set to warn }`() {
+        // Given
+        stubGradleBuildFromResourceFile("build_with_check_deps_set_to_warn.gradle.kts")
+
+        // When
+        val result = GradleRunner.create()
+            .withProjectDir(testProjectDir)
+            .withArguments("assemble")
+            .withPluginClasspath(getTestConfigurationClasspath())
+            .build()
+
+        // Then
+        assertThat(result.task(":assemble")?.outcome).isEqualTo(TaskOutcome.SUCCESS)
+    }
+
+    @Test
+    fun `M fail W assemble { Datadog SDK not in deps }`() {
+        // Given
+        stubGradleBuildFromResourceFile("build_without_datadog_dep.gradle.kts")
+        // When
+        assertThatThrownBy {
+            GradleRunner.create()
+                .withProjectDir(testProjectDir)
+                .withArguments("assemble")
+                .withPluginClasspath(getTestConfigurationClasspath())
+                .build()
+        }.isInstanceOf(UnexpectedBuildFailure::class.java)
+    }
+
+    // endregion
+
+    // region Upload
+
+    @ParameterizedTest
+    @MethodSource("uploadTasks")
+    fun `M success W upload`(taskName: String) {
+        // Given
+        stubGradleBuildFromResourceFile("build_with_datadog_dep.gradle.kts")
+
+        // When
+        val result = GradleRunner.create()
+            .withProjectDir(testProjectDir)
+            .withArguments(taskName)
+            .withPluginClasspath(getTestConfigurationClasspath())
+            .build()
+
+        // Then
+        assertThat(result.task(":$taskName")?.outcome)
+            .isEqualTo(TaskOutcome.SUCCESS)
+    }
+
+    @ParameterizedTest
+    @MethodSource("uploadTasks")
+    fun `M success W upload { custom remote repos url }`(taskName: String) {
+        // Given
+        stubGradleBuildFromResourceFile("build_with_custom_remote_repos_url.gradle.kts")
+
+        // When
+        val result = GradleRunner.create()
+            .withProjectDir(testProjectDir)
+            .withArguments(taskName)
+            .withPluginClasspath(getTestConfigurationClasspath())
+            .build()
+
+        // Then
+        assertThat(result.task(":$taskName")?.outcome)
+            .isEqualTo(TaskOutcome.SUCCESS)
+    }
+
+    @ParameterizedTest
+    @MethodSource("uploadTasks")
+    fun `M success W upload { checkProjectDependecies disabled }`(taskName: String) {
```

Comment:
I think this `checkProjectDependencies` has no impact on the upload task, so we can skip tests with that maybe?